crates/router/src/connector/fiserv.rs

```diff
@@ -61,16 +99,53 @@ impl api::ConnectorCommon for Fiserv {
     fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
         connectors.fiserv.base_url.as_ref()
     }
+    fn get_auth_header(
+        &self,
+        auth_type: &types::ConnectorAuthType,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let auth: fiserv::FiservAuthType = auth_type
+            .try_into()
+            .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+        Ok(vec![(headers::API_KEY.to_string(), auth.api_key)])
+    }
+    fn build_error_response(
+        &self,
+        res: types::Response,
+    ) -> CustomResult<types::ErrorResponse, errors::ConnectorError> {
+        let response: fiserv::ErrorResponse = res
+            .response
+            .parse_struct("Fiserv ErrorResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+
+        let fiserv::ErrorResponse { error, details } = response;
+
+        let message = match (error, details) {
+            (Some(err), _) => err
+                .into_iter()
+                .map(|v| v.message)
+                .collect::<Vec<_>>()
+                .join(""),
+            (None, Some(err_details)) => err_details
+                .into_iter()
+                .map(|v| v.message)
+                .collect::<Vec<_>>()
+                .join(""),
```

Comment:
```suggestion
            (Some(err), _) => err.into_iter().map(|v| v.message).collect::<String>(),
            (None, Some(err_details)) => err_details
                .into_iter()
                .map(|v| v.message)
                .collect::<String>(),
```

Just a minor optimisation, not required to be added here, but consider this for further use-cases